packages/database/src/migrations/20210907022500-ConvertExistingReportsToNewTransformSyntax-modifies-data.js

```diff
@@ -57,29 +66,51 @@ const convertAggregateToGroupRows = oldConfig => {
     ...columnsToAggregate
   } = oldConfig;
   const newConfig = {
-    transform: 'groupRows',
-    where,
+    transform: 'mergeRows',
     title,
     description,
   };
 
+  if (where) {
+    newConfig.where = `=${where}`;
+  }
+
+  const columnsWithBaseStrategy = [];
   Object.entries(columnsToAggregate).forEach(([column, aggregateFunc]) => {
     const isGroupColumn = aggregateFunc === 'group';
     if (isGroupColumn) {
-      if (Array.isArray(newConfig.by)) {
-        newConfig.by = [...newConfig.by, column];
-      } else if (newConfig.by) {
-        newConfig.by = [newConfig.by, column];
+      if (Array.isArray(newConfig.groupBy)) {
+        newConfig.groupBy = [...newConfig.groupBy, column];
+      } else if (newConfig.groupBy) {
+        newConfig.groupBy = [newConfig.groupBy, column];
       } else {
-        newConfig.by = column;
+        newConfig.groupBy = column;
       }
     } else {
       const columnKey = column === '...' ? '*' : column;
-      const mergeStrategy = aggregateFunc === 'drop' ? 'exclude' : aggregateFunc;
-      if (typeof newConfig.mergeUsing === 'object' && newConfig.mergeUsing !== null) {
-        newConfig.mergeUsing = { ...newConfig.mergeUsing, [columnKey]: mergeStrategy };
+      const mergeStrategy = mergeStrategyNames[aggregateFunc]
+        ? mergeStrategyNames[aggregateFunc]
+        : aggregateFunc;
+
+      if (typeof newConfig.using === 'object' && newConfig.using !== null) {
+        newConfig.using = { ...newConfig.using, [columnKey]: mergeStrategy };
+      } else if (typeof newConfig.using === 'string' && newConfig.using !== undefined) {
+        if (mergeStrategy === newConfig.using) {
+          columnsWithBaseStrategy.push(columnKey);
+        } else {
+          const commonStrategy = newConfig.using;
+          const usingObject = Object.fromEntries(
+            columnsWithBaseStrategy.map(columnWithBaseStrategy => [
+              columnWithBaseStrategy,
+              commonStrategy,
+            ]),
+          );
+          usingObject[columnKey] = mergeStrategy;
+          newConfig.using = usingObject;
+        }
       } else {
-        newConfig.mergeUsing = { [columnKey]: mergeStrategy };
+        columnsWithBaseStrategy.push(columnKey);
+        newConfig.using = mergeStrategy;
```

Comment:
Very nice work condensing these down where possible!

Reply:
I was just being lazy before, it wasn't too bad in the end. Thanks for pushing for it 🙌